### ISSUE TITLE
Missing pagination for search results

### ISSUE THREAD

**Issue by svanderwulp** [ucfopen/canvasapi#598]
Endpoint [search recipients](https://canvas.instructure.com/doc/api/search.html#method.search.recipients) is paginated, but corresponding method returns Python list with just the first page of the result.
I expected it to return a PaginatedList, but maybe the lists item type isn't easy to implement (either user of context)?

**Comment by bennettscience** [comment]
If you look at the network tab for a request, is a `Link` header included? The problem is that not _all_ results are paginated. According to the docs, if you search using a `user_id`, you only get one response, never a paginated response.

[Here's the relevant method](https://github.com/ucfopen/canvasapi/blob/f4e725a15a6b9149aba2e74214410c192dfd6d13/canvasapi/canvas.py#L1333-L1350) on the `canvas` object.

**Comment by svanderwulp** [comment]
A Link header is included in the response. I am willing to fix the missing pagination, but unsure how to go about creating a PaginatedList which might contain multiple types of instances (User, Context). When I limit the search results to objects of type 'user' (as per description in API documentation), I am able to get a fully functional PaginatedList.

Maybe there is an example of a PaginatedList list containing multiple object types elsewhere in the code I could use as an example for a fix?

**Comment by bennettscience** [comment]
Yeah, this is a known issue with `PaginatedList` (see #175, #114, and #498). @jessemcbride also [mentions it as a requirement](https://gist.github.com/jessemcbride/133312e5af9e6eb1de7d5b77dc9b7ba9) on #114.

I think the closest you could get is to use the method from #498 where the `GradeChangeEvent` object incudes an `event_type` attribute that can be filtered after collecting the `PaginatedList` response, but I'm not sure what kind of precedent that would set for the rest of the library.

This would also affect #175.
